.github/workflows/ci.yaml

```diff
@@ -0,0 +1,48 @@
+name: Run codebase checks and tests
+
+on: [push, pull_request]
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+
+    services:
+      db:
+        image: postgres:11
+        ports: ["5432:5432"]
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    env:
+      MIX_ENV: test
+      DATABASE_URL: postgres://postgres:postgres@localhost/elixir_boilerplate_test
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-elixir@v1
+        with:
+          elixir-version: 1.10.1
+          otp-version: 22.2 
+      - uses: actions/cache@v1
+        id: cache-mix
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}/mix.lock', github.workspace)) }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12.15
+      - uses: actions/cache@v1
+        id: cache-npm
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles(format('{0}/assets/package-lock.json', github.workspace)) }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+      - run: make dependencies 
+      - run: make check
+      - run: make lint
+      - run: make test
```

Comment:
Shall we not keep using the script `ci-check.sh` instead of each step in different runs like that?

Reply:
Individual steps makes a way better output in the github action UI. Plus, you know in a glimpse what failed or not. Our ci-check script was great with travis which doesn't have a great UI for outputs, but isn't necessary with github actions.

Reply:
_^this_ is my take on the _future_ of `ci-checks.sh` too!

In the end, it would be kept for _development workflow_ purposes, but I personally think the developer "APIs" should all be in `make` and we already maintain the `check` target for that matter…

Let's see what others think before removing it…

Reply:
Summoning @simonprev @remi @garno @ging @lucbernard…